### PR TITLE
GPU Discovery

### PR DESCRIPTION
**Pull request by QuietSwami**
- Added Nvidia Runtime discovery
- Added Dockerfile

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM ubuntu:18.04
```
Comment (cjdcordeiro) [inline Dockerfile]: please use python:3-alpine

Comment (QuietSwami) [reply Dockerfile]: okay, will change

**Review by cjdcordeiro — CHANGES_REQUESTED** [review]
missing compose file

also, please structure the root directory similarly to the other NB components, so that we can clearly identify the code assets and GitHub/Documentation related assets

**Comment by QuietSwami** [comment]
> missing compose file
> 
> also, please structure the root directory similarly to the other NB components, so that we can clearly identify the code assets and GitHub/Documentation related assets

Updated with folder structure and docker compose file. Is the version file, in the src folder, needed?

**Comment by cjdcordeiro** [comment]
@QuietSwami yes please add it

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+version: "3.7"
+
+services:
+  peripheral-manager-gpu:
+    build: .
+    restart: on-failure
+    volumes:
+      - /etc/:/etc/
```
Comment (cjdcordeiro) [inline docker-compose.yml]: missing the :ro

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    environment:
```
Comment (cjdcordeiro) [inline docker-compose.yml]: why do you need these env vars?

File: docker-compose.yml (hunk 2 above)
Comment (QuietSwami) [reply docker-compose.yml]: added

File: docker-compose.yml (hunk 3 above)
Comment (QuietSwami) [reply docker-compose.yml]: I don't know. I copied the one in modbus, and adapted. is now removed

**Comment by QuietSwami** [comment]
> @QuietSwami yes please add it

Is it the same as the one in modbus?

**Comment by cjdcordeiro** [comment]
@QuietSwami yes

File: docker-compose.yml (hunk 3 above)
Comment (QuietSwami) [reply docker-compose.yml]: We need those env variables to access Nuvla through the API

Comment (cjdcordeiro) [reply docker-compose.yml]: not anymore. Check the agent API to know how to register the peripherals

**Review by cjdcordeiro — APPROVED** [review]

**Pull request merged**